### PR TITLE
HHH-14739 - Implement ILike support

### PR DESCRIPTION
**Pull request by jrenaat**
Signed-off-by: Jan Schatteman <jschatte@redhat.com>

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmTreePrinter.java [hunk 1]
```diff
@@ -838,8 +838,9 @@ public Object visitBetweenPredicate(SqmBetweenPredicate predicate) {
 
 	@Override
 	public Object visitLikePredicate(SqmLikePredicate predicate) {
+		String likeType = predicate.isCaseSensitive() ? "like" : "ilike";
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmTreePrinter.java:841]:
```suggestion
		final String likeType = predicate.isCaseSensitive() ? "like" : "ilike";
```

**Review by beikov — CHANGES_REQUESTED** [review]

**Review by beikov — APPROVED** [review]

**Comment by gavinking** [comment]
How many databases does this work on?

**Comment by beikov** [comment]
> How many databases does this work on?

It will work on all databases because we can emulate it with `LOWER(lhs) LIKE LOWER(rhs)`. Also see `org.hibernate.dialect.Dialect#getLowercaseFunction` which was used in the legacy Criteria implementation.

File: documentation/src/main/asciidoc/userguide/chapters/query/hql/HQL.adoc [hunk 2]
```diff
@@ -1780,6 +1791,20 @@ include::{sourcedir}/HQLTest.java[tags=hql-like-predicate-escape-example]
...
+
+Performs a case-insensitive like comparison on string values. The syntax is:
+
+[[hql-ilike-predicate-bnf]]
+[source, JAVA, indent=0]
+----
+include::{extrasdir}/predicate_ilike_bnf.txt[]
+----
+
+The semantics are identical to those of the aforementioned <<hql-like-predicate>>, with the sole difference that the comparison is now case insensitive. Note that for this to work the database will have to be set to a case sensitive collation.
```
Comment (beikov) [inline documentation/src/main/asciidoc/userguide/chapters/query/hql/HQL.adoc:1805]: >Note that for this to work the database will have to be set to a case sensitive collation.

Well, that is not really true I think. The case insensitive comparison works regardless of the collation. The case sensitive like is sensible to the collation though.

Comment (jrenaat) [reply documentation/src/main/asciidoc/userguide/chapters/query/hql/HQL.adoc:1805]: Yeah, you're right, i'll scratch this.

**Review by beikov — APPROVED** [review]

**Review by beikov — APPROVED** [review]

**Pull request merged**